src/Language/DifferentialDatalog/OVSDB/Compile.hs

```diff
@@ -306,10 +316,11 @@ uuidCol = TableColumn { columnPos       = nopos
                       , columnMutable   = Nothing
                       }
 
-tableGetNonROCols :: (?outputs::[(String, [String])]) => Table -> [TableColumn]
+tableGetNonROCols :: (?outputs::[OutputRelConfig]) => Table -> [TableColumn]
 tableGetNonROCols t =
     case lookup (name t) ?outputs of
-         Nothing -> ovscols
-         Just ro -> filter (\col -> notElem (name col) ro) ovscols
+         Nothing         -> ovscols
+         Just (Left ro)  -> filter (\col -> notElem (name col) ro) ovscols
+         Just (Right rw) -> filter (\col -> elem (name col) rw || (name col == "_uuid")) ovscols
```

Comment:
this thing about uuid does not seem to be documented.
Also, the name of this function has a negation, it would be nicer if it was GetRWCols instead.

Reply:
> this thing about uuid does not seem to be documented.

done

> Also, the name of this function has a negation, it would be nicer if it was GetRWCols instead.

there's actually a subtle difference in this case: NonRO means that the column is not labeled as read-only. It can still be read-only if it comes from a read-only table.